name.abuchen.portfolio.tests/src/name/abuchen/portfolio/money/CurrencyUnitTest.java

```diff
@@ -0,0 +1,199 @@
+package name.abuchen.portfolio.money;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CurrencyUnitTest
+{
+
+    private static List<CurrencyUnit> currencyUnits;
+
+    // This method will be run once before all tests to load the available
+    // currency units
+    @BeforeClass
+    public static void setUp()
+    {
+        // Load all available CurrencyUnit instances
+        currencyUnits = CurrencyUnit.getAvailableCurrencyUnits();
+    }
+
+    /**
+     * Test to ensure that there are no duplicate currency codes in the system.
+     * <p>
+     * This test iterates through all available currency units and checks if
+     * there are any duplicate currency codes. A HashSet is used to track unique
+     * codes, and if a duplicate is found, the test will fail and print the
+     * duplicate.
+     */
+    @Test
+    public void testNoDuplicateCurrencyCodes()
+    {
+        Set<String> currencyCodes = new HashSet<>();
+        for (CurrencyUnit currencyUnit : currencyUnits)
+        {
+            var currencyCode = currencyUnit.getCurrencyCode();
+
+            // Using a HashSet to ensure there are no duplicates.
+            assertTrue("Duplicate currency code found: " + currencyCode, currencyCodes.add(currencyCode));
+        }
+    }
+
+    /**
+     * Test to ensure that there are no duplicate currency symbols in the
+     * system.
+     * <p>
+     * This test checks if any currency units share the same currency symbol. It
+     * skips null values (if any) and uses a HashSet to guarantee uniqueness. If
+     * a duplicate is found, the test will fail and print the duplicate symbol.
+     */
+    @Test
+    public void testNoDuplicateCurrencySymbols()
+    {
+        Set<String> currencySymbols = new HashSet<>();
+        for (CurrencyUnit currencyUnit : currencyUnits)
+        {
+            var currencySymbol = currencyUnit.getCurrencySymbol();
+            var currencyCode = currencyUnit.getCurrencyCode();
+            if (currencySymbol != null)
+            {
+                // Use HashSet to ensure there are no duplicates
+                var isAdded = currencySymbols.add(currencySymbol);
+
+                // Using a HashSet to ensure there are no duplicates.
+                assertTrue("Duplicate currency symbol found for currency code " + currencyCode + ": " + currencySymbol,
+                                isAdded);
+            }
+        }
+    }
```

Comment:
I think this test is wrong, because we should allow multiple symbols.
The symbols could be used to display it.

I think for the importer, one has to understand from the context of the document which currency we are talking about. For example, if you just read a $ and it is from the United states, then it is the USD. But we cannot make that assumption for Canadian documents.